compiler/src/org.graalvm.compiler.hotspot.aarch64/src/org/graalvm/compiler/hotspot/aarch64/AArch64HotSpotLIRGenerator.java

```diff
@@ -547,4 +550,27 @@ public SaveRegistersOp createZapRegisters(Register[] zappedRegisters, JavaConsta
     public LIRInstruction createZapArgumentSpace(StackSlot[] zappedStack, JavaConstant[] zapValues) {
         throw GraalError.unimplemented();
     }
+
+    @Override
+    public void emitZeroMemory(Value address, Value length) {
```

Comment:
Is `length` required to be 8 times?

Reply:
Yes. The runtimeAssert in NewObjectSnippets.java has guaranteed that the `size` and `offset` are 8-byte aligned. So the length value is always a multiple of 8. It should be better to add some runtime assertions here. But I don't know if it's possible to do it in this kind of macroAssembler code.

Reply:
It is impossible to do it here. Please add some documentation at both `AArch64ZeroMemoryOp` and this method to warn others that want to reuse these code.